### ISSUE TITLE
Corporate Disclosure Statement: DISC Event Code Changes

### ISSUE THREAD

**Issue by cholly75** [ustaxcourt/ef-cms#8410]
As a Court user, so that DAWSON reflects changes to the Ownership Disclosure Form, I need to update verbiage and behavior associated with the DISC event code.

With the approval of the upcoming rules package by the Court, the Ownership Disclosure Statement (ODS) is being renamed/replaced as the Corporate Disclosure Statement. We need to update all appropriate areas in DAWSON to reflect this change.

This story will address the behavior of the DISC event code (Ownership Disclosure Statement).

We anticipate these changes will need to be live in mid-March 2023.

## Acceptance Criteria
* DISC event code document title is updated to "Corporate Disclosure Statement"
* Document dropdown menus for filing/adding documents reflect new document title of "Corporate Disclosure Statement"
* Cover sheet generated for DISC reflects "Corporate Disclosure Statement"
* Behavior for adding DISC documents to docket remains the same - no changes aside from title/display
* Currently docketed/filed DISC entries are unaffected by this change (no need to make retroactive)

## Notes
* Check on Order for ODS - is this a specific event code as well?
  - YES - there is an OODS - will enter additional story to address

## Story Definition of Ready (updated on 12/23/22)
The following criteria must be met in order for the user story to be picked up by the Flexion development team.
The user story must: 
- [ ]  Is framed in business/user need, the value has been addressed.
- [ ] Includes acceptance criteria
- [ ] Has been refined
- [ ] Pre conditions have been satisfied. 
**Process:** 
Flexion developers and designers will test if the story meets acceptance criteria and test cases in Flexion dev and staging environments (“standard testing”). If additional acceptance criteria or testing scenarios are discovered while the story is in progress, a new story should be created, added to the backlog and prioritized by the product owner. 

## Definition of Done (Updated 5-19-22)
**Product Owner**
 - [ ]  Acceptance criteria have been met and validated on the Court's migration environment
 - [ ] Add scenario to testing document, if applicable (https://docs.google.com/spreadsheets/d/1FUHKC_YrT-PosaWD5gRVmsDzI1HS_U-8CyMIb-qX9EA/edit?usp=sharing)

**UX**
 - [ ] Business test scenarios have been refined to meet all acceptance criteria 
 - [ ] Usability has been validated
 - [ ] Wiki has been updated (if applicable) 
 - [ ] Story has been tested on a mobile device (for external users only)

**Engineering**
 - [ ] Automated test scripts have been written, including visual tests for newly added PDFs
 - [ ] Field level and page level validation errors (front-end and server-side) integrated and functioning
 - [ ] Verify that language for docket record for internal users and external users is identical
 - [ ] New screens have been added to pa11y scripts
 - [ ] All new functionality verified to work with keyboard and macOS voiceover https://www.apple.com/voiceover/info/guide/_1124.html 
 - [ ] READMEs, other appropriate docs, JSDocs and swagger/APIs fully updated
 - [ ] UI should be touch optimized and responsive for external only (functions on supported mobile devices and optimized for screen sizes as required)
 - [ ] Interactors should validate entities before calling persistence methods
 - [ ] Code refactored for clarity and to remove any known technical debt
 - [ ] Acceptance criteria for the story has been met
 - [ ] If there are special instructions in order to deploy into the next environment, add them as a comment in the story
 - [ ] Reviewed by UX on a deployed environment.
 - [ ] Reviewed by PO on a deployed environment. Can be deployed to the Court's `test` environment if prod-like data is required. Otherwise deployed to any `experimental` environment.
 - [ ] Deployed to the Court's `staging` environment.

**Comment by ttlenard** [comment]
## Test Cases

#### 1) Docket Clerk uploads/scans a Corporate Disclosure Statement to a case; New document title is available in the dropdown menu for Document Type.

- Log in as a docket clerk
- Navigate to a case that has a Petitioner with Paper Service
- Click on the Create button
- Select the Add Paper filing link
- Upload or scan in a document
- Fill out the date received
- Click on the Document Type dropdown.
- Scroll or start typing to get to the value: Corporate Disclosure Statement

Expected Results:
- [x] **NEW FUNCTIONALITY** - Corporate Disclosure Statement is now an available option in the dropdown picklist
- [x] **NEW FUNCTIONALITY** - Ownership Disclosure statement is no longer an option in the picklist


#### 2) Docket Clerk adds a Corporate Disclosure Statement as a docket entry; Service Confirmation popup indicates the correct document title.

- Continuing from the previous test
- Select the petitioner from the Who is filing this document section
- Click Save and Serve

Expected Results:
- [x] **NEW FUNCTIONALITY** - The "Are you ready to Initiate Service?" confirmation pop-up modal displays the correct document title of Corporate Disclosure Statement



#### 3) After Confirming Service, Print for paper service screen appears.

- Continuing from the previous test
- Click Yes, Serve


Expected Results
- [x] Print for paper service screen appears for paper petitioner(s)



#### 4) After completing the print for paper service, clerk receives confimration banner.

- Continuing from the previous test
- Click on the Done button

Expected Results
- [x] Green confirmation banner indicating that the entry has been added to the docket record



#### 5) Docket Record displays the appropriate Document Event Code and document title.

- Continuing from the previous test
- Scroll down on the docket record so that you can see the document details that you just added to the docket record

Expected Results:
- [x] **NEW FUNCTIONALITY** - The event code is DISC
- [x] **NEW FUNCTIONALITY** - The document title is Corporate Disclosure Statement
- [x] The rest of the document data is correct (index number, date filed, number of pages, filer, served date, service, edit/seal links are available)



#### 6) When viewing the document, the coversheet of the Corporate Disclosure Statement reflects the appropriate document title

- Continuing from the previous test
- Click on the document link

Expected Results:
- [x] **NEW FUNCTIONALITY** - Document title is Corporate Disclosure Statement



#### 7) Court User navigates to a case that has an existing Ownership Disclosure Statement Document on the Docket Record; Old title of the document remains on the docket record; Coversheet title of the document is still Ownership Disclosure Statement.

- Navigate to a case that has an Ownership Disclosure statement already on the docket record

Expected Results:
- [x] The title of the document is still Ownership Disclosure Statement
- [x] After clicking on the document link to view the document, the coversheet still indicates that the document title is Ownership Disclosure Statement.


#### 8) Docket Clerk edits an existing Ownership Disclosure Statement document type to a Corporate Disclosure Statement; System updates the docket record appropriately.

- Log in as a docket clerk
- Navigate to a case that has an Ownership Dislosure Statement on the docket record.
- Click on the Edit link from the docket record
- From the document type dropdown, select Corporate Disclosure Statement
- Click Save

Expected Results:
- [x] Green confirmation banner indicating that the Docket entry changes were saved.
- [x] Event Code remains the same DISC
- [x] **NEW FUNCTIONALITY** - Document title has been updated to Corporate Disclosure Statement
- [x] **NEW FUNCTIONALITY** - After opening the document, a new coversheet has been added to the document, with a coversheet title of Corporate Disclosure Statement.


~#### 9) Regression test this with the Feature Flag turned off.~

**Comment by rachelschneiderman** [comment]
@codyseibert and I implemented the changes for **this story, ustaxcourt/ef-cms#8407 and ustaxcourt/ef-cms#8409** on a [branch in Flexion's fork](https://github.com/flexion/ef-cms/tree/9827-story).

We decided not to implement a feature flag because the amount of files this change required would make the feature flag expensive and time consuming to produce. Instead, the changes will wait on the branch above until the court approves the new rules package and then get deployed without a feature flag, this was approved by @cholly75 .

**Comment by rachelschneiderman** [comment]
This story has two deploy requirements: 

1. A migration script was added and will need to be run: `0007-update-corporate-disclosure-document.js`
2. The court will need to publish new documents so that the following link is no longer a 404: https://www.ustaxcourt.gov/resources/forms/Corporate_Disclosure_Statement_Form.pdf. We found three links that previously pointed to `Ownership_Disclosure_Statement_Form_6.pdf` which will need to exist on the court's website before these stories are deployed. Those links can be found in: `BeforeStartingCase.jsx` (line 219), `WhatToExpect.jsx` (line 123), `StartCaseStep3.jsx` (line 396)
